typescript/cli/src/commands/validator.ts

```diff
@@ -49,3 +64,60 @@ const addressCommand: CommandModuleWithContext<{
     process.exit(0);
   },
 };
+
+const preFlightCheckCommand: CommandModuleWithContext<{
+  chain: string;
+  validators: string;
+}> = {
+  command: 'preflightCheck',
+  describe: 'Check the validator has announced correctly for a given chain',
+  builder: {
+    chain: makeOptionRequired(chainCommandOption),
+    validators: validatorCommandOption,
+  },
+  handler: async ({ context, chain, validators }) => {
+    const { multiProvider, registry } = context;
+
+    // validate chain
+    if (!multiProvider.hasChain(chain)) {
+      errorRed(`❌ ${chain} is not supported by the current configuration`);
+      process.exit(1);
+    }
+
+    const chainMetadata = await registry.getChainMetadata(chain);
+    if (!chainMetadata) {
+      errorRed(`\n❌ Unable to fetch metadata for chain ${chain}. Exiting.`);
+      process.exit(1);
+    }
```

Comment:
You don't need this check. If the MP knows the chain, then it must have it's chain metadata.

It's fine to query the registry if you prefer but you can also get the metadata synchronously from the MP via `multiprovider.getChainMetadata(chain)`